### ISSUE TITLE
DBs in the "Real World"

### ISSUE THREAD

**Issue by mindovermiles262** [TheOdinProject/rails_course#13]
One of the strengths of TOP is that you're coding locally in your own environment.  We do a great job with this except for one area in which I've heard a lot of people complain about. SQL

I know that we use several online resources and they are fun (SqlZoo, for example), but we should add a component that utilizes Postgres or MySQL locally.  At this point in the curriculum the user should have some background knowledge of setting up programs on their machine.

Practice writing raw SQL queries, setting up users, tables, etc is an important component of working with Databases and Rails in general.

Keep up the strong work, gents 👍

**Comment by CouchofTomato** [comment]
Thanks Andy.

It's certainly something that can be considered. We need to be careful though that we don't bloat the track too much.

One suggestion currently being considered is having a main path which would be similar to what we have with the objective of getting people job ready and then some optional content separately for those looking to take it a step further and dig a little deeper.

**Comment by mindovermiles262** [comment]
I agree this should be an extra activity. Its non-essential with Rails abstracting so much, but still a pain point and something we should add

**Comment by KevinMulhern** [comment]
Completely agree with this, A expanded db section is something I think Odin needs badly for the reasons you mentioned and to have more flexibility with the curriculum. To get rid of the rails portion in web dev 101 we will need to remove the sql lesson as well.

I agree with Austin that it shouldn't bloat the Rails course. I think it should be in its own "mini" course, that just includes one section that teaches just the very basics, at least to begin with. We can reuse and expand on the sql lessons we already have and will include a more broken down lesson structure and get the students using dbs on their local machine.

We can then have this a small course as a prerequisite for the rails course.
